alfresco-bulk-export/src/main/java/org/alfresco/extensions/bulkexport/controler/Engine.java

```diff
@@ -201,47 +177,43 @@ private void storeNodeListToCache(NodeRef rootNode, List<NodeRef> list) throws E
     }
 
 
-
     /**
      * Creates Thread Pool and Tasks with dispatch nodes
      *
      * @param nodesToExport
      */
-    private void exportNodes(final List<NodeRef> nodesToExport) throws InterruptedException, ExecutionException
-    {
+    private void exportNodes(final List<NodeRef> nodesToExport) throws InterruptedException, ExecutionException {
         ExecutorService threadPool = Executors.newFixedThreadPool(nbOfThreads);
-        CompletionService<String> pool = new ExecutorCompletionService<String>(threadPool);
 
-        int previousLowerLimitNodeNumber = 0 ;
-        for(int taskNumber = 1; taskNumber <= this.nbOfTasks; taskNumber++) {
+        int previousLowerLimitNodeNumber = 0;
+        int noOfTasks = new Double(Math.ceil(nodesToExport.size() / this.exportChunkSize)).intValue();
+
+        log.info("Number of tasks: " + noOfTasks);
+
+        for (int taskNumber = 1; taskNumber <= noOfTasks; taskNumber++) {
             int upperLimitNodeNumber = calculateNextUpperLimitNodeNumber(previousLowerLimitNodeNumber, nodesToExport.size());
             int lowerLimitNodeNumber = calculateNextLowerLimitNodeNumber(previousLowerLimitNodeNumber, upperLimitNodeNumber);
-            log.info("Task number"+ taskNumber +" LowerLimitNodeNumber " + lowerLimitNodeNumber);
-            log.info("Task number"+ taskNumber +" UpperLimitNodeNumber " + upperLimitNodeNumber);
+            log.info("Task number" + taskNumber + " LowerLimitNodeNumber " + lowerLimitNodeNumber);
+            log.info("Task number" + taskNumber + " UpperLimitNodeNumber " + upperLimitNodeNumber);
 
             previousLowerLimitNodeNumber = upperLimitNodeNumber;
 
             List<NodeRef> nodesForCurrentThread = nodesToExport.subList(lowerLimitNodeNumber, upperLimitNodeNumber);
-            pool.submit(new NodeExportTask(nodesForCurrentThread, exportVersions, revisionHead, dao, fileFolder, taskNumber));
+            threadPool.submit(new NodeExportTask(nodesForCurrentThread, exportVersions, revisionHead, dao, fileFolder, taskNumber));
         }
+    }
 
-        for(int i = 0; i < this.nbOfTasks; i++){
-            String result = pool.take().get();
-            log.info(result);
+    private int calculateNextLowerLimitNodeNumber(int previousLowerLimitNodeNumber, int upperLimitNodeNumber) {
+        int nextLowerLimitNodeNumber = previousLowerLimitNodeNumber;

```

Comment:
I think it could use:
return previousLowerLimitNodeNumber > upperLimitNodeNumber ? upperLimitNodeNumber :  previousLowerLimitNodeNumber;